selfhost/typechecker.jakt

```diff
@@ -7751,6 +7751,11 @@ struct Typechecker {
                 .error("Can't match on 'void' type", checked_expr.span())
             }
             else => {
+                let is_boolean_match = type_to_match_on is Bool
+                mut seen_true = false
+                mut seen_false = false
+                mut catch_all_span: Span? = None
```

Comment:
This is perfectly fine for now, but going further...we should probably move the value exhaustiveness check somewhere else to open space for other types.